test/trn_func_tests/test_delete_vpc_change_switch_router.py

```diff
@@ -0,0 +1,122 @@
+        #       |   |            |  |                     |   |            |  |
+        #       +---|            |--+                     +---|            |--+
+        #           |            |                            |            |
+        #           |            |                            |            |
+        #           +------------+                            +------------+
+        #           |Transit XDP |                            |Transit XDP |
+        #           |    eth0    |                            |    eth0    |
+        #           +------------+                            +------------+
+        #                  |              +------------+             |
+        #                  |              |  Router1   |             |
+        #                  |              |  Router2   |             |
+        #                  |              |  Router3   |             |
+        #                  |              |            |             |
+        #                  |              |    VPC3    |             |
+        #                  +--------------|            |-------------+
+        #                                 |            |
+        #                                 |            |
+        #                                 +------------+
+        #                                 |Transit XDP |
+        #                                 |    eth0    |
+        #                                 +------------+
+
+        # a. Switch 3, Switch6 and Router3 are all hosted on the same
+        # host
+        # b. Switch 2 is hosted on D3
+        # c. Switch 5 is hosted on D2
+
+        self.droplets = {
+            "d1": droplet("d1"),
+            "d2": droplet("d2"),
+            "d3": droplet("d3"),
+            "d4": droplet("d4"),
+            "d5": droplet("d5"),
+            "switch-1": droplet("switch-1"),
+            "switch-4": droplet("switch-4"),
+            "router-1": droplet("router-1"),
+            "router-2": droplet("router-2"),
+        }
+
+        self.c = controller(self.droplets)
+
+        self.c.create_vpc(3, cidr("16", "10.0.0.0"), ["router-1"])
+
+        self.c.create_network(3, 10, cidr("24", "10.0.0.0"), ["switch-1"])
+
+        self.c.create_network(3, 20, cidr("24", "10.20.0.0"), ["switch-4"])
+
+        self.ep1 = self.c.create_simple_endpoint(3, 10, "10.0.0.2", "d1")
+        self.ep2 = self.c.create_simple_endpoint(3, 10, "10.0.0.3", "d2")
+
+        self.ep3 = self.c.create_simple_endpoint(3, 20, "10.20.0.4", "d3")
+        self.ep4 = self.c.create_simple_endpoint(3, 20, "10.20.0.5", "d4")
+
+        self.c.add_router(3, "router-2")
+        self.c.add_switch(3, 10, "d3")
+        self.c.add_switch(3, 10, "d5")
+        self.c.add_switch(3, 20, "d2")
+        self.c.add_switch(3, 20, "d5")
+        self.c.add_router(3, "d5")
+        self.c.remove_switch(3, 20, "d2")
+        self.c.remove_router(3, "d5")
+
+        self.c.delete_vpc(3, cidr("16", "10.0.0.0"))
```

Comment:
The VPC id is unique in the test, do you need to pass the cidr as well?

Reply:
I don't. Fixing now.